orca-core-tck/src/main/groovy/com/netflix/spinnaker/orca/pipeline/persistence/PipelineExecutionRepositoryTck.groovy

```diff
@@ -550,6 +550,28 @@ abstract class PipelineExecutionRepositoryTck<T extends ExecutionRepository> ext
     status << ExecutionStatus.values()
   }
 
+  def "should return task ref for currently running pipeline by correlation id"() {
```

Comment:
I see that this test nicely fails when I run
```
$ ./gradlew orca-redis:test --tests JedisPipelineExecutionRepositorySpec
```
without the redis fix, and passes with it.

But, neither
```
$ ./gradlew orca-sql:test --tests **MySqlPipelineExecutionRepositorySpec
```
nor
```
$ ./gradlew orca-sql:test --tests PgSqlPipelineExecutionRepositorySpec
```
fail without the SqlExecutionRepository change.  Is there some other test that demonstrates the bug?

Reply:
The occurrence of the "duplicate entry exception for correlation_ids table" bug arises when two or more concurrent instances of the Orca service attempt to write identical "correlation id" simultaneously. Presently, all existing tests are executed sequentially within a single thread, thereby preventing the replication of this bug. However, it is essential to note that these tests primarily serve the purpose of verifying the correctness of the business logic.

Reply:
I have reservations regarding our ability to construct a unit test capable of adequately cover this particular bug scenario. Even in the event that we endeavor to establish several distinct processes executed concurrently on a CPU with multiple cores, achieving simultaneous data write operations in 100% of cases appears unattainable.

Reply:
It does seem like there'd be a way to demonstrate that the old code was busted by writing two pipelines with the same correlation id in sequence.  Granted, it's more difficult to verify that the new code is working correctly, and I agree throwing a bunch of threads at it hoping isn't great.  If there's a way to spy on `onConflict` or `doNothing`, we could potentially do it, but you've convinced me that we're ok without an explicit test for this.